plexus-compilers/plexus-compiler-eclipse/src/main/java/org/codehaus/plexus/compiler/eclipse/EclipseJavaCompiler.java

```diff
@@ -91,31 +94,28 @@ public CompilerResult performCompile( CompilerConfiguration config )
     {
         List<CompilerMessage> errors = new LinkedList<CompilerMessage>();
 
-        List<String> classpathEntries = config.getClasspathEntries();
+        File outFile = new File(config.getOutputLocation());
 
-        URL[] urls = new URL[1 + classpathEntries.size()];
-
-        int i = 0;
-
-        try
-        {
-            urls[i++] = new File( config.getOutputLocation() ).toURL();
-
-            for ( String entry : classpathEntries )
-            {
-                urls[i++] = new File( entry ).toURL();
-            }
-        }
-        catch ( MalformedURLException e )
-        {
-            throw new CompilerException( "Error while converting the classpath entries to URLs.", e );
+        if(!outFile.isDirectory() && !outFile.mkdirs()) {
+            throw new CompilerException( "Error creating output directory: " + outFile.getAbsolutePath());
         }
 
-        ClassLoader classLoader = new URLClassLoader( urls );
+        List<String> cpe = new ArrayList<String>();
+        cpe.addAll(Arrays.asList(System.getProperty("sun.boot.class.path").split(":")));
```

Comment:
Well, it'll fail to recognize packages like `java.lang` and such that are provided by the runtime otherwise, things that are in `rt.jar`.